File: Algorithms/Sorting_Algorithms/Bucket_Sort/BucketSort.java
// Bucket sort in Java

import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class BucketSort {
  public void bucketSort(float[] arr, int n) {
    if (n <= 0)
      return;
    @SuppressWarnings("unchecked")
    ArrayList<Float>[] bucket = new ArrayList[n];

    // Create empty buckets
    for (int i = 0; i < n; i++)
      bucket[i] = new ArrayList<Float>();

    // Add elements into the buckets
    for (int i = 0; i < n; i++) {
      int bucketIndex = (int) arr[i] * n;
      bucket[bucketIndex].add(arr[i]);
    }

    // Sort the elements of each bucket
    for (int i = 0; i < n; i++) {
      Collections.sort((bucket[i]));
    }

    // Get the sorted array
    int index = 0;
    for (int i = 0; i < n; i++) {
      for (int j = 0, size = bucket[i].size(); j < size; j++) {
        arr[index++] = bucket[i].get(j);
      }
    }
  }

  // Driver code
  public static void main(String[] args) {
    BucketSort b = new BucketSort();
    Scanner s = new Scanner(System.in);  // Create a Scanner object
    System.out.println("Enter no of elemets");
    int n=s.nextInt();
    ArrayList<Float>[] arr = new ArrayList[n];
    for(int i=0;i<n;i++){
      System.out.println("Enter a number");
      float num=s.nextFloat();
      num.add(arr[]);
    }
    
    b.bucketSort(arr, n);

    for (float i : arr)
      System.out.print(i + "  ");
  }
}

File: Algorithms/Sorting_Algorithms/Bucket_Sort/bucketsort.php
<?php
function insertion_sort(&$elements, $fn = 'comparison_function') {
  if (!is_array($elements) || !is_callable($fn)) return;
  for ($i = 1; $i < sizeof($elements); $i++) {
    $key = $elements[$i]; 
    $j = $i - 1; // this will be in $b in comparison function
    while ( $j >= 0 && $fn($key, $elements[$j]) ) {
      $elements[$j + 1] = $elements[$j]; // shift right
      $j = $j - 1;
    }
    $elements[$j + 1] = $key;
  }
}

function comparison_function(&$a, &$b) {
  return $a < $b;
}

function bucket_sort(&$elements) {
  $n = sizeof($elements);
  $buckets = array();
  // Initialize the buckets.
  for ($i = 0; $i < $n; $i++) {
    $buckets[$i] = array();
  }
  // Put each element into matched bucket.
  foreach ($elements as $el) {
    array_push($buckets[ceil($el/10)], $el);
  }
  // Sort elements in each bucket using insertion sort.
  $j = 0;
  for ($i = 0; $i < $n; $i++) {
    // sort only non-empty bucket
    if (!empty($buckets[$i])) {
      insertion_sort($buckets[$i]);
      // Move sorted elements in the bucket into original array.
      foreach ($buckets[$i] as $el) {
        $elements[$j++] = $el;
      }
    }
  }
}
$a = array(-1,0,10,15,-2);
bucket_sort($a); // Sort the elements
echo "\nSorted Array :\n";
var_dump($a);
?>
